### PR TITLE
[ML] [Data Frame] add support for weighted_avg agg

### PR DESCRIPTION
**Pull request by benwtrent**
Simple enough change, `weighted_avg` support follows naturally with the support of `avg` and dynamically determined mappings. 

Since `weighted_avg` is a `MultiValuesSourceAggregationBuilder` that allows for scripted internals (instead of `field` references), and a `format` parameter, we should fall back for the mapping to be `dynamic`. This allows for the end user to supply whatever mapped type they want in a created index/template or for ES to determine the appropriate mapping dynamically.

### CODE REVIEW

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by dimitris-athanasiou** [comment]
I'm not sure but could there be a documentation change pending to list the new supported aggregation?

**Comment by benwtrent** [comment]
@dimitris-athanasiou there probably should be a list of supported aggregations...Not sure where that lives in the docs though.

File: x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFramePivotRestIT.java [hunk 1]
```diff
@@ -473,6 +473,45 @@ public void testPivotWithGeoCentroidAgg() throws Exception {
...
+        setupDataAccessRole(DATA_ACCESS_ROLE, REVIEWS_INDEX_NAME, dataFrameIndex);
+
+        final Request createDataframeTransformRequest = createRequestWithAuth("PUT", DATAFRAME_ENDPOINT + transformId,
+            BASIC_AUTH_VALUE_DATA_FRAME_ADMIN_WITH_SOME_DATA_ACCESS);
+
+        String config = "{"
+            + " \"source\": {\"index\":\"" + REVIEWS_INDEX_NAME + "\"},"
+            + " \"dest\": {\"index\":\"" + dataFrameIndex + "\"},";
+
+        config += " \"pivot\": {"
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFramePivotRestIT.java]: No need to do concatenation, can be done in declaration of "String config".

**Review by przemekwitek — APPROVED** [review]
LGTM

Question for my education:
I'm wondering why do we have to whitelist particular aggregation types. Can't pivot work with all aggregations? If not, why?

**Comment by benwtrent** [comment]
@przemekwitek the reason for whitelisting is that there are many multi-bucket aggs, pipeline aggs, and a couple of metric aggs that complicate parsing the results to put in a doc.

**Comment by przemekwitek** [comment]
> @przemekwitek the reason for whitelisting is that there are many multi-bucket aggs, pipeline aggs, and a couple of metric aggs that complicate parsing the results to put in a doc.

Ok, thanks @benwtrent

**Pull request merged**